src/2982R0_std_quantity_as_a_numeric_type.md

```diff
@@ -2318,40 +2328,43 @@ This has significant advantages.  It preserves the _form_ of the code, making th
 than replacement with a function such as `is_gt_zero`.  It also reduces the number of new comparison APIs
 a user must learn: `Zero` handles them all.
 
-`Zero` has a few downsides though:
-
-1. CTAD can't deduce the quantity type as there is no unit provided. So if we try to refactor the
-   following:
-
-    ```cpp
-    quantity<si::metre / si::second> q1 = 0 * m / s;  // OK
-    quantity q2 = 0 * m / s;                          // OK
-    ```
-
-   only the first one will work and the second one will fail to compile:
+`Zero` has one downside: it will not work when passed across _generic quantity_ interfaces.  `Zero`'s value
+comes in situations where the surrounding context makes it unambiguous which quantity type it should
+construct.  While it converts to any _specific_ quantity type, it is not itself a quantity.  This could
+confuse users.
 
-    ```cpp
-    quantity<si::metre / si::second> q1 = ZERO;  // OK
-    quantity q2 = ZERO;                          // Compile-time error
-    ```
+This downside manifests in several different ways.  Here are some examples:
 
-2. While refactoring the [@MP-UNITS] code to try out this approach we found out also some other places
+1. While refactoring the [@MP-UNITS] code to try out this approach we found out a perfectly reasonable place
    where we could not replace numerical value `0` with `ZERO`:
 
    ```cpp
-   for (auto tt = 0 * ms; tt <= 50 * ms; ++tt) { /* ... */ }
+   msl_altitude alt = mean_sea_level + 0 * si::metre;  // OK
+   msl_altitude alt = mean_sea_level + ZERO;           // Compile-time error
    ```
 
-   ```cpp
-   msl_altitude alt = mean_sea_level + 0 * si::metre; 
-   ```
+   This would not work because the `mean_sea_level` is an absolute point origin that stores the information
+   about the quantity type but not its value and unit.
 
-   The second example, would not work because the `mean_sea_level` is an absolute point origin that
-   stores the information about the quantity type but not its value and unit.
+2. Callsites passing `ZERO` can add friction when refactoring a concrete interface to be more generic.
 
-3. It works for addition but not multiplication as we do not know if we should multiply by a scalar or
-   a quantity. In the latter case, we do not have enough information to produce not only a unit but even
-   quantity type and dimension of the result:
+    ```cpp
+    namespace v1 { void foo(quantity<si::metre> q); }
+    namespace v2 { void foo(QuantityOf<isq::length> auto q); }
```

Comment:
Adding namespaces actually makes it harder to understand the refactoring issue as those are separate functions possibly doing something completely different.

Reply:
That was even more true before, with `foo` and `boo` being _completely_ different functions.  I think if `v1` and `v2` are the names of the namespaces, then it evokes the "refactoring" use case more clearly than before.

Maybe make one an `inline` namespace?  I don't know; I think it's fine how it is.